### PR TITLE
Add long error explanation for E0576

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #61137.

### CODE REVIEW

File: src/librustc_resolve/error_codes.rs [hunk 1]
```diff
@@ -1611,6 +1611,31 @@ fn print_on_failure(state: &State) {
...
+```compile_fail,E0576
+trait Hello {
+    type Who;
+
+    fn hello() -> <Self as Hello>::You; // error!
+}
+```
+
+In this example, we tried to use the non-existent associated type `You` of the
+`Hello` trait. To fix this error, just use an existing associated type:
```
Comment (matthewjasper) [inline src/librustc_resolve/error_codes.rs]:
```suggestion
`Hello` trait. To fix this error, use an existing associated type:
```

**Review by matthewjasper — COMMENTED** [review]
r=me with comment addressed

File: src/librustc_resolve/error_codes.rs (hunk 1 above)
Comment (kinnison) [reply src/librustc_resolve/error_codes.rs]: r=me if this suggestion is adopted.

**Comment by GuillaumeGomez** [comment]
Since CI passed the first time, I guess I can just r+ it...

@bors: r=matthewjasper,kinnison

**Comment by GuillaumeGomez** [comment]
And here we go again...

@bors: r=matthewjasper,kinnison

**Comment by GuillaumeGomez** [comment]
Rebased.

**Pull request merged**